Documentation/writing-subplugin-tensor-filter.md

```diff
@@ -8,9 +8,60 @@ You can support a new neural-network framework (e.g., Tensorflow and Caffe) or a
 
 It is called "**subplugin**" because it is a plugin for a GStreamer plugin, ```tensor_filter```.
 
+**A subplugin should NEVER require properties of input/output dimensions or types.** You should use such properties for pipeline debugging and validation and the subplugin should fetch such information directly from model files via framework APIs and GSTCAPS negotiations.
 
 
-## Quick guide on writing a tensor\_filter subplugin for a new framework/hardware.
+
+# Quick guide on writing a tensor\_filter subplugin for a new framework/hardware.
+
+The most recommended method is to write a C++ tensor\_filter subplugin as a derived class of ```tensor_filter_subplugin``` class [/gst/nnstreamer/include/nnstreamer_cppplugin_api_filter.hh].
+Then, you can register the derived class, usually by calling ```tensor_filter_subplugin::register_subplugin()``` in the init function so that subplugin infrastructure recognizes your subplugin whtn the shared library is loaded.
```

Comment:
```suggestion
Then, you can register the derived class, usually by calling ```tensor_filter_subplugin::register_subplugin()``` in the init function so that subplugin infrastructure recognizes your subplugin when the shared library is loaded.
```

Reply:
Fixed.